File: GXpert/GXpert.Web/Modules/Activation/ActivationLog/ActivationLogColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Activation.Columns;

[ColumnsScript("Activation.ActivationLog")]
[BasedOnRow(typeof(ActivationLogRow), CheckNames = true)]
public class ActivationLogColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string SerialKey { get; set; }
    public string TeacherPrn { get; set; }
    public string PlayListTitle { get; set; }
    public string DeviceId { get; set; }
    public string DeviceDetails { get; set; }
    [QuickFilter]
    public short EStatus { get; set; }
    public string Note { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/ActivationLogColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EKeyStatus } from "../Web/Enums.EKeyStatus";
import { ActivationLogRow } from "./ActivationLogRow";

export interface ActivationLogColumns {
    Id: Column<ActivationLogRow>;
    Code: Column<ActivationLogRow>;
    SerialKey: Column<ActivationLogRow>;
    TeacherPrn: Column<ActivationLogRow>;
    PlayListTitle: Column<ActivationLogRow>;
    DeviceId: Column<ActivationLogRow>;
    DeviceDetails: Column<ActivationLogRow>;
    EStatus: Column<ActivationLogRow>;
    Note: Column<ActivationLogRow>;
}

export class ActivationLogColumns extends ColumnsBase<ActivationLogRow> {
    static readonly columnsKey = 'Activation.ActivationLog';
    static readonly Fields = fieldsProxy<ActivationLogColumns>();
}

[EKeyStatus]; // referenced types